core/rpc/core/src/impl/utils.rs

```diff
@@ -489,141 +408,126 @@ impl<C: CkbRpc> MercuryRpcImpl<C> {
-                    IdentityFlag::Ethereum => Ok(self.get_builtin_script(PW_LOCK, pubkey_hash)),
-                    _ => Err(CoreError::UnsupportIdentityFlag.into()),
-                }
-            }
+    fn get_default_owner_lock_by_script(
+        &self,
+        script: packed::Script,
+    ) -> InnerResult<packed::Script> {
+        let lock_args = script.args().raw_data();
+        if self.is_script(&script, SECP256K1)? || self.is_script(&script, ACP)? {
+            let args = H160::from_slice(&lock_args[0..20]).expect("Impossible: parse args");
+            Ok(self.get_builtin_script(SECP256K1, args))
+        } else if self.is_script(&script, PW_LOCK)? {
+            let args = H160::from_slice(&lock_args[0..20]).expect("Impossible: parse args");
+            Ok(self.get_builtin_script(PW_LOCK, args))
+        } else {
+            Err(CoreError::UnsupportAddress.into())
+        }
+    }
 
-            Item::Address(addr) => {
-                let addr =
-                    parse_address(&addr).map_err(|e| CoreError::CommonError(e.to_string()))?;
-                let script = address_to_script(addr.payload());
-                let lock_args = script.args().raw_data();
-                if self.is_script(&script, SECP256K1)? || self.is_script(&script, ACP)? {
-                    let args = H160::from_slice(&lock_args[0..20]).expect("Impossible: parse args");
-                    Ok(self.get_builtin_script(SECP256K1, args))
-                } else if self.is_script(&script, PW_LOCK)? {
-                    let args = H160::from_slice(&lock_args[0..20]).expect("Impossible: parse args");
-                    Ok(self.get_builtin_script(PW_LOCK, args))
-                } else {
-                    Err(CoreError::UnsupportAddress.into())
-                }
-            }
+    async fn get_lock_by_out_point(
+        &self,
+        out_point: packed::OutPoint,
+    ) -> InnerResult<packed::Script> {
+        let mut cells = self
+            .storage
+            .get_cells(
+                Context::new(),
+                Some(out_point),
+                vec![],
+                vec![],
+                None,
+                PaginationRequest::default(),
+            )
+            .await
+            .map_err(|e| CoreError::DBError(e.to_string()))?;
 
-            Item::Record(id) => {
-                let (_, ownership) = decode_record_id(id)?;
-                match ownership {
-                    Ownership::Address(address) => {
-                        Ok(self.get_default_lock_by_item(Item::Address(address))?)
-                    }
-                    Ownership::LockHash(_) => Err(CoreError::UnsupportOwnership.into()),
-                }
-            }
+        if cells.response.is_empty() {
+            return Err(CoreError::CannotFindDetailedCellByOutPoint.into());
         }
+
+        Ok(cells.response.swap_remove(0).cell_output.lock())
```

Comment:
`cells.response` has been checked as non-empty in previous line, please use index here and remove the mut keyword:
```
Ok(cells.response[0].cell_output.lock())
```